remote.go

```diff
@@ -849,17 +861,12 @@ func getHavesFromRef(
 		return nil
 	}
 
-	// No need to load the commit if we know the remote already
-	// has this hash.
-	if remoteRefs[h] {
-		haves[h] = true
-		return nil
-	}
```

Comment:
Would this not cause a performance regression, as we may be increasing the memory churn?